client/pdo/client/scripts/UpdateCLI.py

```diff
@@ -97,20 +98,38 @@ def LocalMain(config, message) :
         sys.exit(-1)
 
     # ---------- set up the enclave service ----------
-    try :
-        enclave_url = service_config['PreferredEnclaveService']
-        enclave_client = EnclaveServiceClient(enclave_url)
-        logger.info('contact enclave service at %s', enclave_url)
-    except KeyError as ke :
-        logger.error('missing configuration parameter %s', str(ke))
-        sys.exit(-1)
-    except Exception as e :
-        logger.error('unable to connect to enclave service; %s', str(e))
-        sys.exit(-1)
+    if service_config.get('EnclaveServiceNames'): #use the database to get the enclave for the contract
+        logger.info('Using eservice database to look up service URL for the contract enclave')
+        try:
+            eservice_to_use = random.choice(service_config['EnclaveServiceNames'])
+            # load the eservice database
+            if os.path.exists(service_config['EnclaveServiceDatabaseFile']):
+                try:
+                    db.load_database(service_config['EnclaveServiceDatabaseFile'])
+                except Exception as e:
+                    logger.error('Error loading eservice database %s', str(e))
+                    sys.exit(-1)
+            enclave_client = db.get_client_by_name(eservice_to_use)
+        except Exception as e:
+            logger.error('Unable to get the eservice client using the eservice database: %s', str(e)) 
+            sys.exit(-1)   
+    else:
+        try:
+            enclave_url = service_config['PreferredEnclaveService'] 
+        except Exception as e:
+            logger.error('missing configuration parameter %s', str(ke))
+            sys.exit(-1)
+        try :
+            enclave_client = EnclaveServiceClient(enclave_url)
+        except Exception as e :
+            logger.error('unable to connect to enclave service; %s', str(e))
+            sys.exit(-1)
+        
```

Comment:
again... why not just add these back into the database so that others can use them? you don't have to store the database.

Reply:
same answer as for createCLI. I agree we should add this feature to shell, but may be not for one time runs